### PR TITLE
Added dihedral fits tests and updated/fixed build errors in the docs

### PR DESCRIPTION
**Pull request by bc118**
- Added the rest of the newly created examples tests and additional tests to catch errors.  
- Updated the docs text and removed errors and warnings when building the docs.

### CODE REVIEW

File: docs/getting_started/quick_start/fit_a_dihedral_with_Gaussian_and_write_all_files.rst [hunk 1]
```diff
@@ -21,7 +44,7 @@ Import the required packages.
 .. code:: ipython3
 
     import unyt as u
-    from mosdef_dihedral_fit.dihedral_fit.fit_dihedral_with_gomc import fit_dihedral_with_gomc
+    #from mosdef_dihedral_fit.dihedral_fit.fit_dihedral_with_gomc import fit_dihedral_with_gomc
```
Comment (daico007) [inline docs/getting_started/quick_start/fit_a_dihedral_with_Gaussian_and_write_all_files.rst:47]: should we just remove this line?

**Review by daico007 — APPROVED** [review]
LGTM

File: docs/getting_started/quick_start/fit_a_dihedral_with_Gaussian_and_write_all_files.rst (hunk 1 above)
Comment (bc118) [reply docs/getting_started/quick_start/fit_a_dihedral_with_Gaussian_and_write_all_files.rst:47]: I actually left it this way, thinking people will only copy each example.

Comment (bc118) [reply docs/getting_started/quick_start/fit_a_dihedral_with_Gaussian_and_write_all_files.rst]: why are we removing it, do we not need it?

**Pull request merged**